src/CovidSim.cpp

```diff
@@ -5311,16 +5311,16 @@ void RecordInfTypes(void)
 		{
 			//				i=Cells[b].members[c];
 			if (j == 0) j = k = Households[Hosts[i].hh].nh;
-			if ((Hosts[i].inf != InfStat_Susceptible) && (Hosts[i].inf != InfStat_ImmuneAtStart))
+			if (Hosts[i].is_infected())
 			{
 				if (Hosts[i].latent_time * P.TimeStep <= P.SampleTime)
 					TimeSeries[(int)(Hosts[i].latent_time * P.TimeStep / P.SampleStep)].Rdenom++;
 				infcountry[Mcells[Hosts[i].mcell].country]++;
-				if (abs(Hosts[i].inf) < InfStat_Recovered)
+				if (Hosts[i].has_not_recovered())
 					l = -1;
 				else if (l >= 0)
 					l++;
-				if ((l >= 0) && ((Hosts[i].inf == InfStat_RecoveredFromSymp) || (Hosts[i].inf == InfStat_Dead_WasSymp)))
+				if (l >= 0 && ((Hosts[i].has_recovered() || Hosts[i].is_dead()) && Hosts[i].is_symptomatic()))
```

Comment:
Could this be simplified? Can we check logic has not changed from original?

Reply:
Logic has not changed from the original. I triple checked each conversion.
I believe a lot of the old logic can be simplified, but I didn't want to risk getting it wrong, and breaking things.

My approach is to first abstract the logic into functions, then split the underlying enum into separate variables, THEN simplify the logic. Simplifying the logic on a few variables is going to be a lot easier than simplifying the logic on a range on INT values.